$MirOS: ports/net/centericq/patches/patch-src_icqface_cc,v 1.3 2005/10/09 22:07:43 tg Exp $
--- src/icqface.cc.orig	Sun Aug 28 01:32:59 2005
+++ src/icqface.cc	Wed Feb  1 21:22:22 2006
@@ -97,7 +97,7 @@ const char *strint(unsigned int i) {
     static char buf[64];
 
     if(i) {
-	sprintf(buf, "%lu", i);
+	snprintf(buf, sizeof(buf), "%u", i);
     } else {
 	buf[0] = 0;
     }
@@ -230,7 +230,10 @@ void icqface::showtopbar() {
 
     attrset(conf.getcolor(cp_status));
     mvhline(0, 0, ' ', COLS);
-    mvprintw(0, 0, _(" CENTERICQ %s  UNSENT: %lu"), VERSION, em.getunsentcount());
+#ifndef PORT_VERSION
+#define PORT_VERSION VERSION
+#endif
+    mvprintw(0, 0, _(" CENTERICQ %s  UNSENT: %lu"), PORT_VERSION, em.getunsentcount());
     mvprintw(0, COLS-buf.size()-1, "%s", buf.c_str());
 }
 
@@ -1256,7 +1259,7 @@ void icqface::inforss(dialogbox &db, icq
     char buf[512];
 
     if(mi.checkfreq) {
-	sprintf(buf, _("%lu minutes"), mi.checkfreq);
+	sprintf(buf, _("%lu minutes"), (unsigned long)mi.checkfreq);
 	freq = buf;
 	if(mi.checklast) next = strdateandtime(mi.checklast+mi.checkfreq*60);
 	    else next = _("Now");
